samza-core/src/test/java/org/apache/samza/operators/impl/TestOperatorImpl.java

```diff
@@ -47,10 +48,12 @@
 
 public class TestOperatorImpl {
   private Context context;
+  private JobContextMetadata jobContextMetadata;
 
   @Before
   public void setup() {
     this.context = new MockContext();
+    this.jobContextMetadata = new JobContextMetadata(null, null);
```

Comment:
Consider mocking this object with Mockito instead of using a real instance. It makes the test a bit more resilient to changes to JobContextMetadata in the future.